assignment5.txt

```diff
@@ -0,0 +1,30 @@
+Caleb Gershengorn
+
+1:
+You would call the function by typing fxn(a,b,c) where a,b,c would be constants or variables inserted by the user.
+It might look something like this:
+void main(){
+	printf("enter a number\n");
+	scanf("%d",&a); \\These two lines take in an int as 'a'
+	printf("enter a float\n");
+	scanf("%f",&b);  \\These two lines take in an float as 'b'
+	printf("enter another number\n");
+	scanf("%d",&c); \\These two lines take in an int as 'c'
+	printf("%d\n", fxn(a,b,c); \\calls the fxn function and prints the result using %d because fxn calls for an int
+	\\This would call the fxn and use the three values the user entered to run the function
+	}
+
+2:
+Iteration is running a loop a certain number of times in order to achieve a certain result, updating the value of a variable every time the loop runs and returning a new value when the loop runs.
+Recursion is a process that involves a function calling itself a certain number of times, updating a counting variabl but not updating the variable that you want to change until the end.
+Once the recursive function reaches its end, it remembers how many times it has run, and runs itself that many times, updating the variable each time.
+Loops are preferable in instances when the variables are simple to sort through, but recursion is useful when you might have to check a certain value again or when something is a function of itself, like a tree.
+Recursion is often not preferable because calling a function takes more space in memory than running an iteration, making it less efficient usually.
```

Comment:
right, complexity is key! and now that we're learning about memory management and data structures, you can further understand this by understanding that each time a function is called, a stack is initialized. So having a recursive function can be very memory ineffective.